### PR TITLE
[refactor](functions) clean some ip functions code and make IS_IP_ADDRESS_IN_RANGE DEPENDS_ON_ARGUMENT

### PR DESCRIPTION
**Pull request by zclllyybb**
## Proposed changes

Issue Number: close #xxx

docs pr: https://github.com/apache/doris-website/pull/680
## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by zclllyybb** [comment]
run buildall

**Comment by zclllyybb** [comment]
wait https://github.com/apache/doris/pull/35312 merge first

File: be/src/vec/functions/function_ip.h [hunk 1]
```diff
@@ -137,13 +139,7 @@ static inline bool try_parse_ipv4(const char* pos, Int64& result_value) {
...
 template <IPConvertExceptionMode exception_mode, typename ToColumn>
 ColumnPtr convert_to_ipv4(ColumnPtr column, const PaddedPODArray<UInt8>* null_map = nullptr) {
-    const ColumnString* column_string = check_and_get_column<ColumnString>(column.get());
-
-    if (!column_string) {
-        throw Exception(ErrorCode::INVALID_ARGUMENT,
-                        "Illegal column {} of argument of function {}, expected String",
-                        column->get_name());
-    }
+    const auto* column_string = check_and_get_column<ColumnString>(column.get());
```
Comment (Mryange) [inline be/src/vec/functions/function_ip.h:142]: If nullptr is not checked here, then may be using assert cast？

Comment (zclllyybb) [reply be/src/vec/functions/function_ip.h]: good advise. fixed it.

**Comment by zclllyybb** [comment]
run buildall

**Review by zhangstar333 — APPROVED** [review]

**Review by Gabriel39 — APPROVED** [review]

**Pull request merged**